tests/python/pants_test/pantsd/test_process_manager.py

```diff
@@ -28,13 +29,24 @@ def fake_process(**kwargs):
     return proc
 
 
+# For history of this test's flakiness: https://github.com/pantsbuild/pants/issues/6836
+def test_deadline_until(caplog):
+    pm = ProcessManager("_test_", metadata_base_dir=safe_mkdtemp())
+    with pytest.raises(ProcessManager.Timeout):
+        with caplog.at_level(logging.INFO):
+            pm._deadline_until(
+                lambda: False, "the impossible", "completed? how?", timeout=0.5, info_interval=0.1
+            )
+    assert 4 <= len(caplog.records) <= 6, f"Expected between 4 and 6 infos, got: {caplog.records}"
```

Comment:
This test is absolutely still flaky. If you look at the code under test it just snaps time and sleeps. Arbitrary time can pass between the 1st snap and the 1st time check / sleep under load. That directly leads to the <4 condition in the OP. So, by inspection, this is simply a bad test. It either needs to be completely re-written using new techniques or else go back to being dead. As it stands it's just a time bomb again.

Reply:
Yeah, on closer inspection, 100%. Almost any test that counts real elapsed sleep time is a time bomb (no pun intended). This either needs an injected mock clock, or to be deleted. But being turned off but remaining in the codebase for 5 years seems like the worst of all worlds.

Reply:
https://github.com/pantsbuild/pants/pull/19850